src/Orleans/Core/GrainAttributes.cs

```diff
@@ -248,14 +248,38 @@ namespace Providers
         /// </para>
         /// </summary>
         [AttributeUsage(AttributeTargets.Class)]
-        public sealed class StorageProviderAttribute : Attribute
+        public sealed class StorageProviderAttribute : PersistenceProviderAttribute
         {
             public StorageProviderAttribute()
             {
-                    ProviderName = Runtime.Constants.DEFAULT_STORAGE_PROVIDER_NAME;
+                ProviderName = Runtime.Constants.DEFAULT_STORAGE_PROVIDER_NAME;
             }
+        }
+
+
+        /// <summary>
+        /// The [Orleans.Providers.LogViewProvider] attribute is used to specify a log view provider for grains that extend LogViewGrain&lt;T&gt;.
+        /// <para>
+        /// If present, the [Orleans.Providers.LogViewProvider] attribute overrides any [Orleans.Providers.StorageProvider] attribute,
+        /// otherwise we try to find a storage provider as for all other grains. 
+        /// </para>
+        /// </summary>
+        [AttributeUsage(AttributeTargets.Class)]
+        public sealed class LogViewProviderAttribute : PersistenceProviderAttribute
+        {
+            public LogViewProviderAttribute()
+            {
+            }
```

Comment:
They inherit from the same base attribute because they are alternative ways to specify a persistence mechanism for the grain. For log view grains, users can either specify a storage provider or a log view provider. Those are different kinds of persistence providers, but they are somewhat interchangeable (a standard storage provider can be used for log view grains). See the code inside `Catalog.SetupPersistenceProvider` - it finds out which persistence attribute is specified, and retrieves the appropriate type of provider.

There is no default log view provider (because the default is to use the default storage provider), therefore I am not setting the string there.